google/cloud/storage/internal/sha256_hash.cc

```diff
@@ -23,10 +23,11 @@ namespace storage {
 inline namespace STORAGE_CLIENT_NS {
 namespace internal {
 
-std::vector<std::uint8_t> Sha256Hash(std::string const& str) {
+namespace {
+std::vector<std::uint8_t> Sha256Hash(void const* data, std::size_t size) {
```

Comment:
`[unsigned] char const* data`, so that the meaning of `size` is clear?  That is, leave the `void*` stuff to the calls to the lower layers.

Reply:
Then I need to reinterpret cast from either the call with a `char const*` or the call with a `std::uint8_t const*`.    Used a template, and checked the template parameter against the constraints we need, or at least what I think are those constraints.